cpp/memilio/utils/parameter_distribution_wrapper.h

```diff
@@ -0,0 +1,139 @@
+/* 
+* Copyright (C) 2020-2024 MEmilio
+*
+* Authors: Julia Bicker
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#ifndef PARAMETER_DISTRIBUTION_WRAPPER_H
+#define PARAMETER_DISTRIBUTION_WRAPPER_H
+
+#include "memilio/io/io.h"
+#include "memilio/utils/compiler_diagnostics.h"
+#include "parameter_distributions.h"
+
+namespace mio
+{
+
+class ParameterDistributionWrapper
+{
+public:
+    ParameterDistributionWrapper()
+        : m_dist(nullptr)
+    {
+    }
+
+    ParameterDistributionWrapper(ParameterDistribution& dist)
+        : m_dist(std::unique_ptr<ParameterDistribution>(dist.clone()))
+    {
+    }
+
+    ParameterDistributionWrapper(const ParameterDistributionWrapper& other)
+    {
+        m_dist = (other.m_dist == nullptr) ? nullptr : std::unique_ptr<ParameterDistribution>(other.m_dist->clone());
+    }
+
+    ParameterDistributionWrapper(ParameterDistributionWrapper&& other)
+    {
+        m_dist = (other.m_dist == nullptr) ? nullptr : std::unique_ptr<ParameterDistribution>(other.m_dist->clone());
+    }
+
+    ParameterDistributionWrapper& operator=(ParameterDistributionWrapper const& other)
+    {
+        m_dist = (other.m_dist == nullptr) ? nullptr : std::unique_ptr<ParameterDistribution>(other.m_dist->clone());
+        return *this;
+    }
+
+    ParameterDistributionWrapper& operator=(ParameterDistributionWrapper&& other)
+    {
+        m_dist = (other.m_dist == nullptr) ? nullptr : std::unique_ptr<ParameterDistribution>(other.m_dist->clone());
+        return *this;
+    };
+
+    ~ParameterDistributionWrapper() = default;
+
+    std::vector<double> params() const
+    {
+        if (m_dist == nullptr) {
+            log_error("Distribution is not defined. Parameters cannot be deduced.");
+        }
+        return m_dist->params();
```

Comment:
Logging is not enough here, as a nullptr dereference is undefined behaviour. If we are lucky, this return will immediately cause a segfault and crash.